lib/portlayer/exec/container.go

```diff
@@ -475,12 +476,20 @@ func (c *Container) stop(ctx context.Context, waitTime *int32) error {
 				}
 				log.Warnf("invalid power state during power off: %s", terr.ExistingState)
 
+			case *types.GenericVmConfigFault:
+				if terr.Reason == notSuspendedMsg {
+					log.Infof("power off %s task skipped due to guest shutdown", c.ExecConfig.ID)
+					return nil
+				}
+				log.Warnf("hard power off failed due to: %#v", terr)
+
 			default:
 				log.Warnf("hard power off failed due to: %#v", terr)
 			}
 		}
 		c.State = existingState
```

Comment:
@cgtexmex I think this needs to change - perhaaps move this to a `event.Set(state State)`, `cache.Set(state State)`, so that there's a single place where this gets managed.

In that model, this could would call `event.Set(STOPPING)` to report that that's what's being attempted, avoiding the issues with other VM events from vSphere and reported state changes from other operations in the PL overwriting state.

My concern with this is that we could be overwriting a value that is not c.StateStopping (most likely c.Stopped set via events).

Not for this PR, but for reference.